### ISSUE TITLE
ClassCastException

### ISSUE THREAD

**Issue by CharlieZheng** [OneSignal/OneSignal-Android-SDK#1553]
**Description:**

We got a crash, and the method stack below
```
Fatal Exception: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Number
       at com.onesignal.JSONUtils.generateJsonDiff(JSONUtils.java:66)
       at com.onesignal.UserState.generateJsonDiffFromIntoSyncValued(UserState.java:395)
       at com.onesignal.UserStateSynchronizer.updateDeviceInfo(UserStateSynchronizer.java:509)
       at com.onesignal.OneSignalStateSynchronizer.updateDeviceInfo(OneSignalStateSynchronizer.java:222)
       at com.onesignal.OneSignal.registerUserTask(OneSignal.java:1474)
       at com.onesignal.OneSignal.access$1500(OneSignal.java:84)
       at com.onesignal.OneSignal$9.run(OneSignal.java:1441)
       at java.lang.Thread.run(Thread.java:762)
```

**Environment**

1. We using OneSignal v4.6.3
2. we integrate OneSignal by gradle

**Steps to Reproduce Issue:**

we have no idea to reproduce

**Anything else:**

(crash stacktraces, as well as any other information here)

```
Fatal Exception: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Number
       at com.onesignal.JSONUtils.generateJsonDiff(JSONUtils.java:66)
       at com.onesignal.UserState.generateJsonDiffFromIntoSyncValued(UserState.java:395)
       at com.onesignal.UserStateSynchronizer.updateDeviceInfo(UserStateSynchronizer.java:509)
       at com.onesignal.OneSignalStateSynchronizer.updateDeviceInfo(OneSignalStateSynchronizer.java:222)
       at com.onesignal.OneSignal.registerUserTask(OneSignal.java:1474)
       at com.onesignal.OneSignal.access$1500(OneSignal.java:84)
       at com.onesignal.OneSignal$9.run(OneSignal.java:1441)
       at java.lang.Thread.run(Thread.java:762)
```

**Comment by shubhamvashisht** [comment]
@nan-li I don't have access to any samsung device with Android 7.0. we can see thousands of occurrences in the firebase crashlytics for this.

**Comment by shubhamvashisht** [comment]
@nan-li Hi, Any update?
Thanks

**Comment by nan-li** [comment]
The fix is released in OneSignal-Android-SDK [v4.7.2](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/4.7.2).
